pax-runtime/src/engine/mod.rs

```diff
@@ -414,14 +413,47 @@ impl PaxEngine {
         self.runtime_context.node_cache.get(&id)
     }
 
-    pub fn get_focused_element(&self) -> Option<Rc<ExpandedNode>> {
-        let (x, y) = self.runtime_context.globals().viewport.bounds;
-        self.runtime_context
-            .get_topmost_element_beneath_ray(Point2::new(x / 2.0, y / 2.0))
-    }
-
     /// Called by chassis when viewport size changes, e.g. with native window resizes
     pub fn set_viewport_size(&mut self, new_viewport_size: (f64, f64)) {
         self.runtime_context.globals_mut().viewport.bounds = new_viewport_size;
     }
+
+    pub fn global_dispatch_key_down(&self, args: ArgsKeyDown) {
+        self.root_node.recurse_visit_postorder(
```

Comment:
Is there a reason we trigger key handlers on all nodes rather than just the root node?

Reply:
disregard as per discord convo